torch/linalg/__init__.py

```diff
@@ -258,44 +269,49 @@
 """)
 
 eig = _add_docstr(_linalg.linalg_eig, r"""
-linalg.eig(input, *, out=None) -> (Tensor, Tensor)
+linalg.eig(A, *, out=None) -> (Tensor, Tensor)
 
-Computes the eigenvalue decomposition of a square matrix if it exists. Supports batched inputs.
+Computes the eigenvalue decomposition of a square matrix if it exists.
 
-For a square matrix :math:`A`, this is defined as
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+the eigenvalue decomposition of a square matrix
+:math:`A \in \mathbb{K}^{n \times n}` is defined as
 
 .. math::
 
-    A = V \operatorname{diag}(L) V^{-1}
+    A = V \operatorname{diag}(\Lambda) V^{-1}\mathrlap{\qquad V \in \mathbb{C}^{n \times n}, \Lambda in \mathbb{C}^n}
 
-This decomposition exists if and only if :math:`A` is `diagonalizable`_. This is the case when all its eigenvalues are different.
+This decomposition exists if and only if :math:`A` is `diagonalizable`_.
+This is the case when all its eigenvalues are different.
```

Comment:
I'd put this sentence immediately after the previous one. Suggested rewording:

"Equivalently, when the eigenvalues of :math:\`A\` are unique."

Reply:
The problem is that it is not a necessary condition, but only a sufficient one. The identity matrix is trivially diagonalisable but it has repeated eigenvalues.